sdk/mobile-center-push/src/main/java/com/microsoft/azure/mobile/push/Push.java

```diff
@@ -224,4 +273,102 @@ public synchronized void setInstanceEnabled(boolean enabled) {
+    }
+
+    @Override
+    public void onActivityResumed(Activity activity) {
+        checkPushInActivityIntent(activity);
+    }
+
+    @Override
+    public void onActivityPaused(Activity activity) {
+        mActivity = null;
+    }
+
+    /**
+     * Check for push message clicked from notification center in activity intent.
+     *
+     * @param activity activity to inspect.
+     */
+    private synchronized void checkPushInActivityIntent(Activity activity) {
+        mActivity = activity;
+        if (isEnabled() && mInstanceListener != null) {
+            Bundle extras = activity.getIntent().getExtras();
+            if (extras != null) {
+                String googleMessageId = extras.getString(EXTRA_GOOGLE_MESSAGE_ID);
+                if (googleMessageId != null && !googleMessageId.equals(mLastGoogleMessageId)) {
+                    MobileCenterLog.info(LOG_TAG, "Clicked push message from background id=" + googleMessageId);
+                    mLastGoogleMessageId = googleMessageId;
+                    Map<String, String> customData = new HashMap<>();
+                    Map<String, Object> allData = new HashMap<>();
+                    for (String extra : extras.keySet()) {
+                        allData.put(extra, extras.get(extra));
+                        if (!EXTRA_STANDARD_KEYS.contains(extra)) {
+                            customData.put(extra, extras.getString(extra));
+                        }
+                    }
+                    MobileCenterLog.debug(LOG_TAG, "Push intent extra=" + allData);
+                    mInstanceListener.onPushNotificationReceived(activity, new PushNotification(null, null, customData));
+                }
+            }
+        }
+    }
+
+    /**
+     * Called when push message received in foreground.
+     *
+     * @param remoteMessage push message details.
+     */
+    synchronized void onMessageReceived(RemoteMessage remoteMessage) {
+        MobileCenterLog.info(LOG_TAG, "Received push message in foreground id=" + remoteMessage.getMessageId());
+        if (isEnabled() && mInstanceListener != null) {
+            String title = null;
+            String message = null;
+            RemoteMessage.Notification notification = remoteMessage.getNotification();
+            if (notification != null) {
+                title = notification.getTitle();
+                message = notification.getBody();
+            }
+            final PushNotification pushNotification = new PushNotification(title, message, remoteMessage.getData());
+            HandlerUtils.runOnUiThread(new Runnable() {
+
+                @Override
+                public void run() {
+                    deliverForegroundPushNotification(pushNotification);
+                }
+            });
+        }
+    }
+
+    /**
+     * Top level method needed for synchronized code coverage.
+     */
+    private synchronized void deliverForegroundPushNotification(PushNotification pushNotification) {
```

Comment:
Just curious, any reason of taking out as a method? Looks it is used by one place.